pkg/test/echo/server/forwarder/instance.go

```diff
@@ -16,19 +16,22 @@ package forwarder
 
 import (
 	"context"
+	"fmt"
 	"io"
 	"net/http"
 	"time"
 
 	"github.com/golang/sync/errgroup"
+	"golang.org/x/sync/semaphore"
 
 	"istio.io/istio/pkg/test/echo/common"
 	"istio.io/istio/pkg/test/echo/proto"
-	"istio.io/pkg/log"
 )
 
 var _ io.Closer = &Instance{}
 
+const maxConcurrency = 10
```

Comment:
just wondering - did this fix any test failures? Its a good idea but if it fixes things I may be a bit concerned we are masking a problem potentionally?

Reply:
As in we should be able to support a ton of concurrent connections? I'm not really sure how Kubernetes/Docker handles fd limits for network connections or what our settings are like, but we should keep these somewhere reasonable to not exhaust them. 

One problem this could mask is this [Kubernetes issue](https://kubernetes.io/blog/2019/03/29/kube-proxy-subtleties-debugging-an-intermittent-connection-reset/#what-caused-the-connection-reset) with Pod->Service connections. They fixed it by dropping `INVALID` packets in https://github.com/kubernetes/kubernetes/pull/74840Would our iptables interfere. They only effect container->envoy, then kube-proxy rules still apply between envoy->out, right? The failed connections have ClusterIPs in their error messages so I imagine we're still using the kube-proxy NATing.

Reply:
I agree we should not be sending like 1k connections at once, but in theory it should work right? Maybe not in kind but on a real cluster. A user app may do this for example. Yes kube-proxy capture envoy -> out

Reply:
What kind of problem under-test would it be masking? Maybe making this a flag/request param with a default makes sense if we ever want to use this for load testing. Some concurrent conns ~10-50 should be enough to prove things are working.